### PR TITLE
Share `getFileBySourceArchiveName` implementation

### PR DESCRIPTION
**Pull request by hvitved**
Swift is still on its own implementation, since their `File` class does not implement the newly introduced `FileSig` signature (the predicates `getStem` and `getExtension` are missing).

### CODE REVIEW

**Review by erik-krogh — APPROVED** [review]

File: shared/util/codeql/util/FileSystem.qll
Comment (aibaars) [inline shared/util/codeql/util/FileSystem.qll:347]: Why not simply add this definition directly into the `module Make` defined above?

Comment (hvitved) [reply shared/util/codeql/util/FileSystem.qll:347]: I need the `FileSig` signature for another PR anyway, and this way it will not be exposed if a language decides to reexport the result of `Make` directly.

**Review by michaelnebel — APPROVED** [review]
C# LGTM

**Review by owen-mc — APPROVED** [review]
Go LGTM

**Review by jketema — APPROVED** [review]
C++ 👍 

It seems that we probably should just fix up the Swift implementation to conform to the signature, or are there issues with that?

**Comment by hvitved** [comment]
> It seems that we probably should just fix up the Swift implementation to conform to the signature, or are there issues with that?

I'll leave that to @github/codeql-swift . They should consider switching to the shared `FileSystem.qll` implementation.

**Comment by jketema** [comment]
> > It seems that we probably should just fix up the Swift implementation to conform to the signature, or are there issues with that?
> 
> I'll leave that to @github/codeql-swift . They should consider switching to the shared `FileSystem.qll` implementation.

I see that this is blocked on Swift introducing some container concept. Currently only files are extracted, but not folders.

**Comment by aibaars** [comment]
> I see that this is blocked on Swift introducing some container concept. Currently only files are extracted, but not folders.

You don't really need to extract folder, you can also define an IPA type that treats all path-prefixes ending with `/` as folders.

File: shared/util/codeql/util/FileSystem.qll
Comment (aibaars) [reply shared/util/codeql/util/FileSystem.qll:347]: Ok, in isolation it looked a bit overkill to have the `FileSig` and `IdeContextual` modules . However, if you need `FileSig` anyway then this looks fine to me.

**Review by aibaars — APPROVED** [review]

**Pull request merged**
